src/runtime/refiner.ts

```diff
@@ -1621,11 +1621,24 @@ interface OperatorInfo {
 
 const numericTypes: Primitive[] = [Primitive.NUMBER].concat(discreteTypes);
 
+function powBigInt(x: bigint, n: bigint): bigint {
```

Comment:
Probably worth adding some tests for this.

It seems unfortunate that we have to use this polyfill even when it's natively supported by the browser. Is there some way of bypassing Closure's rewrite rules to use the native `**` operator rather than `Math.pow`?

Reply:
There are already some tests for it in the refiner-tests code (i.e. we check the bounds for Long and Int match some known constants which implicitly tests this code).

Maybe we can ask @cromwellian or @shans if they know a way around the Closure compiler for this? (Any ideas Ray / Shane?)

It is unfortunate, but we only use it for calculating these 4 constants, calculated once and they have associated tests.

Reply:
I spent a bit of time digging into this. The Closure rewrite pass is called `rewriteExponentialOperator`. The operator was introduced in ES2016, so we need to pass the `--language_out=ECMASCRIPT_2016` flag to the Closure Compiler and it should stop it from doing this rewrite.

However, the BUILD rules are impossible for me to follow... I can't figure out where to put that flag. @cromwellian might know.

Reply:
Thanks Cameron, I'll angling towards using the constants, though it is unsatisfying, uploading the change now.

Reply:
Normally its something like

js_binary(name = "...", 
  defs = ["flags"],
  ...)

so you'd put the language_out in defs attribute

Reply:
The only this is, we aren't using js_binary (afaict).

This is probably the wrong place to be talking about this though, since this code isn't checked into GitHub